File: 6KYU/Most profit from stock quotes/solution.cs
using System;

static class Kata
{
    public static int GetMostProfitFromStockQuotes(int[] quotes)
    {
        var profit = 0;
        var largest = 0;
      
        for(var i = quotes.Length-1; i >= 0; i--)
        {
          if(quotes[i] < largest)
            profit += largest - quotes[i];
          else
            largest = quotes[i];
        }
        return profit;
    }
}

File: 6KYU/Most profit from stock quotes/solution.go
package kata

func GetMostProfitFromStockQuotes(quotes []int) int {
	largest := 0
	l := len(quotes)
	profit := 0
	for i := l - 1; i >= 0; i-- {
		if quotes[i] < largest {
			profit += largest - quotes[i]
		} else {
			largest = quotes[i]
		}
	}

	return profit
}
